### ISSUE TITLE
Use `intake` GUI search with STAC ItemCollections

### ISSUE THREAD

**Issue by kthyng** [intake/intake#617]
I'm not sure whether this question is for `intake-stac` or `intake` so please help me redirect as necessary.

My workflow is:
1. Set up a STAC catalog for datasets in netcdf files that are time series.
2. Search dynamic STAC catalog using `pystac-client` and return items.
3. Read items in as `intake` catalog using development version of `intake-stac` (so I can use `pystac-client` instead of `sat-search` though the results are the same for this question).
4. Load resulting `intake` catalog into `intake` GUI for browsing. 

The discovered datasets are all sort of flattened into just "items" from their STAC hierarchy of collections with sub-items in step #2. So then in step #4, the datasets come into the `intake` GUI as this "ItemCollection", but not as separate listings within that catalog (see image). This seems to make it so I cannot search the whole catalog of datasets at once for keyword matching.

![image](https://user-images.githubusercontent.com/3487237/135894300-be3dcc12-32d9-48cc-8f28-f366d088eaa2.png)

From the `intake` docs, I believe when I read into `intake`, I should have the resulting datasets instead have the arrangement like the following image in which the dataset items are under "Sources" in the middle column instead of in the left hand side column, which would then allow me to select the catalog in the left hand side column and perform a search for all items in that catalog:
![image](https://user-images.githubusercontent.com/3487237/135894909-cc78ee2e-0e57-4353-81ec-5799d22a90c2.png)

So, my goal is to be able to search all datasets from a STAC catalog at once. I think I'm looking for a way to recategorize the items to be recognized as a "catalog" in the intake GUI. Alternatively, I think the same goal can be completed with a way to search over a selection of entries from the left hand side column. Thank you for any help!

**Comment by martindurant** [comment]
I have no specific STAC knowledge, but the GUI image implies that the ItemCollection is a catalog, containing the many catalogs that you see, which each have item(s). You should be able to select the ItemCollection thing (the one above the upmost `|-----`) and search that; it will produce a new search results catalog containing the items that match.

**Comment by kthyng** [comment]
Yes of course, thank you for the suggestion. By avoiding the GUI, it looks like the search is actually performing ok:

```
cat_search = cat.search('temp', 2)
```
returns the correct items when checked with `list(cat_search)`. 

So that implies I guess that the issue is with the GUI at least for representing the returned search values. Could it be having trouble with the setup of the catalog, related to how the items appear in the left hand column instead of all in the middle column?

I also tried adding `cat_search` to the GUI with 
```
intake.gui.add(cat_search)
intake.gui
```
to try for a workaround but it has the same issue where it only shows a single asset from the search (it looks the same as the previous screenshot).

**Comment by martindurant** [comment]
In `intake.interface.catalog.sarch.Search`, the search is done as 
```
            new_cat = cat.search(self.inputs.text,
                                 depth=self.inputs.depth)
```
where you can find the values of the parameters
```
intake.gui.cat.search.cats
intake.gui.cat.search.inputs.text
intake.gui.cat.search.inputs.depth
```
Can you find check if any of these don't align with `cat.search('temp', 2)`?

Note that the design of the GUI is to have catalogs in the left panel (and a search result is also a catalog) and sources in the middle.
